glib/src/translate.rs

```diff
@@ -2274,6 +2274,440 @@ where
     }
 }
 
+#[derive(Debug, PartialEq, Eq)]
+enum ContainerTransfer {
+    Full,
+    Container,
+    None,
+}
+
+/// Slice of elements of type `T` allocated by the GLib allocator.
+///
+/// This can be used like a `&[T]`.
+pub struct GPtrSlice<T: GlibPtrDefault + FromGlibPtrFull<<T as GlibPtrDefault>::GlibType>> {
```

Comment:
Maybe all this should be moved to some kind of `glib::collections` module and also get re-exported from the crate root?

@bilelmoussaoui @GuillaumeGomez ?

Reply:
What would be the advantage?

Reply:
1. We usually don't have `glib::translate` types in the public API
2. We usually don't have `glib::some_module::Type` in the public API but just `glib::Type`
3. I'm going to add more things in the near future (something around `GSList` / `GList` and `GHashTable`, for example)

Reply:
Sounds reasonable. Please move them into `glib::collections` then! :)

Reply:
Done!